src/core/lib/stripe.ts

```diff
@@ -1,7 +1,188 @@
 import Stripe from "stripe";
 import config from "@config";
+import OptionsService from "@core/services/OptionsService";
+import currentLocale, { locales } from "@locale";
 
-export default new Stripe(config.stripe.secretKey, {
+export const stripe = new Stripe(config.stripe.secretKey, {
   apiVersion: "2023-10-16",
   typescript: true
 });
+
+/**
+ * Get the tax rates by display name
+ * @param displayName The display name
+ * @param params The stripe request options, e.g. `{ limit: 1, active: true }`
+ * @returns The tax rates
+ */
+export const stripeTaxRatesGetByDisplayName = async (
+  displayName: string,
+  params?: Stripe.TaxRateListParams
+) => {
+  return (await stripe.taxRates.list(params)).data.filter(
+    (taxRate) => taxRate.display_name === displayName
+  );
+};
+
+/**
+ * Get the default display name for the tax rate.
+ *
+ * @returns The default display name
+ */
+export const stripeTaxRateGetDefaultDisplayName = () =>
+  (currentLocale().membershipBuilder.steps.joinForm as any).taxRate ||
```

Comment:
I've just realised on testing that this is actually a true type error so shouldn't be cast. It's because some of the locales don't have `taxRate` defined, so this will throw an error if the locale is set to that

Reply:
yes I know, but `joinForm` is set everywhere, only `taxRate` is not. How else do you think I should implement it? I use English as fallback and the code should not lead to an error.

Reply:
I changed the type to ` { taxRate: string | undefined } | undefined` now